public/app/features/trails/interactions.ts

```diff
@@ -0,0 +1,149 @@
+import { AdHocVariableFilter } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+
+import { LayoutType } from './ActionTabs/types';
+import { TrailStepType } from './DataTrailsHistory';
+import { ActionViewType } from './shared';
+
+// prettier-ignore
+type Interactions = {
```

Comment:
I really like the documentation here. It gives me a clear picture of the important interactions and goals of data trails. Nice organization!